### PR TITLE
Dm 4081 move to top pb edit button

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-4081

## Description - what does this code do?
* Adds links to all PB component forms that when clicked will move the component to the top of the list of components in the PB edit page.

## Testing done - how did you test it/steps on how can another person can test it 
1. Create a page group and page
2. Add one of each type of component
3. For each component in the edit page, click the 'move to top' link and verify the component moves to the top of the component list.
4. Save the page and verify the list persists as shown on screen after clicking "Move to Top" for one or more of the components.
5. Create a new component, before saving the page click "Move to Top" button for the new component, verify it moves to the top of the component list. Save the page and verify the ordering persists as shown prior to saving.

## Screenshots, Gifs, Videos from application (if applicable)
Screenshot 2023-08-01 at 1 35 44 PM

## Acceptance criteria
- [ ]

## Definition of done
- [x] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating JIRA issue
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/views/active_admin/resource/_page_accordion_component_form.html.arb [hunk 1]
```diff
@@ -18,6 +18,8 @@ html = Arbre::Context.new do
       end
       if component
         input type: 'hidden', value: component.id, name: "page[page_components_attributes][#{placeholder}][component_attributes][id]"
+
+        a 'Move to Top', href: '#', class: 'move-to-top float-right', id: component.page_component.id
       end
```
Comment (PhilipDeFraties) [inline app/views/active_admin/resource/_page_accordion_component_form.html.arb:23]: One thing I am iffy about is the fact that the `page_components` table is not set up to use uuids meaning the id attributed to the html elements in these links are the actual ids for the component's `PageComponent` relation. Did some research and apparently its not necessarily a security risk? 

If needed I suppose the id attributed to the link can be combination of the component's class and it's uuid, pass that as the request param, then use in the controller action to look up the component in order to access it's `PageComponent` relation to make the ordering update, but I wasn't sure if its actually necessary.

File: app/assets/javascripts/active_admin.js [hunk 2]
```diff
@@ -457,6 +457,21 @@ const pageComponentNames = [
...
+            liElement.parentElement.prepend(liElement);
+
+            Array.from(liElement.parentElement.children).forEach(function (siblingLi, index) {
+                let positionInput = siblingLi.querySelector('input[name$="[position]"]');
+                if (positionInput) {
+                    positionInput.value = index + 1;
+                }
+            });
+        }
+    });
```
Comment (PhilipDeFraties) [inline app/assets/javascripts/active_admin.js:476]: seems so obvious now :man-facepalming:

Comment (PhilipDeFraties) [inline app/assets/javascripts/active_admin.js:476]: seems so obvious now 😂

**Review by camillevilla — APPROVED** [review]
✅ Functionality for this is working great. I think Athena and content editors will be really happy with this feature

For consistency within the JS file, let's move this to a function that clearly describes its intended use, e.g.
```js
function moveToTopBtn(){
        $(document).on('click', '.move-to-top', function (event) {
```

**Pull request merged**